include/blas_helper.cuh

```diff
@@ -552,4 +552,16 @@ namespace quda
 
   } // namespace blas
 
+  template <typename A, typename B> void check_size(const A &a, const B &b)
+  {
+    if (a.size() != b.size()) errorQuda("Mismatched sizes a=%lu b=%lu", a.size(), b.size());
+  }
+
+  template <typename A, typename B, typename... Args> void check_size(const A &a, const B &b, const Args &...args)
```

Comment:
This gives me pause because I think the number of comparisons grows exponentially and has redundant comparisons... I can't quite think of a sane fix, though. What are your thoughts @maddyscientist ?

Reply:
One thing to do here is to remove the redundant comparison.  I think the number of comparisons will now grow linearly.

9881399fdb04f8e06ab9544dc5321a1b80efeb80